Fix typos across the codebase

nathanjmcdougall opened the pull request:
<!--

Thank your for contributing to ObsPy!

!! Please check that you select the **correct base branch** (details see below link) !!

Before submitting a PR, please review the pull request guidelines:
https://github.com/obspy/obspy/blob/master/CONTRIBUTING.md#submitting-a-pull-request

Also, please make sure you are following the ObsPy branching model:
https://github.com/obspy/obspy/wiki/ObsPy-Git-Branching-Model

-->

### What does this PR do?

This PR introduces mostly cosmetic changes to comments to fix typos. There are a few cases of renaming variables and in a couple of cases in `obspy\core\util\testing.py` there appeared be minor bugs due to typos. The main reason I am targeting the maintenance branch rather than `master` is that in some cases, error messages are affected - I am open to rebasing though.

To some extent this is just to begin a discussion about the best way to approach this. It might be better to handle this with multiple separate PRs but I would take advice on that.

I appreciate this affects many files and some of them seem like they ought to be left as-is - happy to revert some, if so.

### Why was it initiated?  Any relevant Issues?

ObsPy is a dependency of USGS's `libcomcat`, and while debugging something I came across a typo. I am a fan of the [codespell](https://github.com/codespell-project/codespell) spellchecker so I thought I would run it across the codebase.

I didn't auto-fix the typos - I manually went through the codebase and verified each one - there are naturally some false positives. The `pyproject.toml` settings I used for codespell were:

```TOML
[tool.codespell]
skip = [
    "*.pdf",
    "*.seed",
    "*.bib",
    "*.css*",
    "*.scss",
    "*/tests/data",
    "./obspy/signal/src/evalresp/ltmain.sh",
    "./obspy/signal/src/evalresp/config.guess",
    "./obspy/signal/src/bbfk.c",
    "./obspy/signal/__init__.py",
    "./obspy/io/xseed/blockette/blockette055.py",
    "./obspy/io/sac/util.py",
    "./obspy/io/segy/tests/test_segy.py",
    "./obspy/io/segy/tests/test_su.py",
    "./obspy/io/pde/docs/mchedr.txt",
    "./obspy/io/nlloc/docs/nlloc_hyp_file.txt",
    "./obspy/io/nied/fnetmt.py",
    "./obspy/io/mseed/src/libmseed/test/read-text-encoded.test.ref",
    "./obspy/io/gse2/src/GSE_UTI/buf.c",
    "./obspy/io/arclink/inventory.py"
]
ignore-words-list = ["aer", "alo", "ccompiler", "BORD", "boun", "equipments", "FELD", "fpt", "fro", "LAUCH", "nd", "NED", "numer", "ot", "parms", "parm", "recend", "SECON", "splitted", "Stange", "STATI", "te", "endianess", "UE", "Whit", "WTH", "wth"]
```


### PR Checklist
- [?] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] While the PR is still work-in-progress, the `no_ci` label can be added to skip CI builds
- [ ] If the PR is making changes to documentation, docs pages can be built automatically.
      Just add the `build_docs` tag to this PR.
      Docs will be served at [docs.obspy.org/pr/{branch_name}](https://docs.obspy.org/pr/) (do not use master branch).
      Please post a link to the relevant piece of documentation.
- [ ] If all tests including network modules (e.g. `clients.fdsn`) should be tested for the PR,
      just add the `test_network` tag to this PR.
- [ ] All tests still pass.
- [ ] Any new features or fixed regressions are covered via new tests.
- [ ] Any new or changed features are fully documented.
- [ ] Significant changes have been added to `CHANGELOG.txt` .
- [?] First time contributors have added your name to `CONTRIBUTORS.txt` .
- [ ] If the changes affect any plotting functions you have checked that the plots
      from all the CI builds look correct. Add the "upload_plots" tag so that plotting 
      outputs are attached as artifacts. 
- [ ] New modules, add the module to `CODEOWNERS` with your github handle
- [ ] Add the yellow `ready for review` label when you are ready for the PR to be reviewed.

ThomasLecocq commented on obspy/io/mseed/src/libmseed/doc/libmseed-UsersGuide:
```diff
@@ -312,7 +312,7 @@ a custom logging facility.  See the man page for more details.
 The default destination for log messages is standard output (stdout),
 while all diagnostic (including error) messages go to standard error
-(stderr).  Most of the internal messages emmited by the library are
+GEODEV(stderr).  Most of the internal messages emitted by the library are
```
check len(line)<80

nathanjmcdougall replied:
That's a mistake anyway (GEODEV is something unrelated to this!)

ThomasLecocq commented on obspy/io/mseed/src/libmseed/doc/ms_readleapseconds.3:
```diff
@@ -14,7 +14,7 @@ ms_readleapseconds - Read a leap second file into a global buffer
 These functions read a leap seconds file and store a list of leap
 seconds in an internal, global list.  The leap seconds are used to
-determine the proper end time of a record (and consquently a trace).
+determine the proper end time of a record (and consequently a trace).
```
check len(line)<80

ThomasLecocq commented on obspy/signal/src/evalresp/configure:
```diff
@@ -13268,7 +13268,8 @@ $debug ||
-ac_cr='
'
+ac_cr='
+'
```
weird stuff here ?

nathanjmcdougall replied:
It looks like the reason that happened is because I've got git configured to censor Windows Carriage Return line endings (^M) with Unix style line endings. I should turn that off for this repo.

I'll try to push a change reverting it.

ThomasLecocq commented on obspy/signal/src/evalresp/ltmain.sh:
```diff
@@ -3058,7 +3058,7 @@ func_to_host_pathlist ()
         # into '.;' and ';.', and winepath ignores them completely.
         func_to_host_pathlist_tmp2="$1"
         # Once set for this call, this variable should not be
-        # reassigned. It is used in tha fallback case.
```
typical @megies gettho slang

ThomasLecocq commented on obspy/signal/src/evalresp/regsub.c:
```diff
@@ -48,7 +48,7 @@ char *dest;
 
 	if (prog == NULL || source == NULL || dest == NULL) {
-		evr_regerror("NULL parm to evr_regsub");
+		evr_regerror("NULL param to evr_regsub");
```
parm vs param : not protected word in C ?

megies commented on obspy/io/gcf/src/gcf_io.c:
```diff
@@ -16,7 +16,7 @@
 //                  blocks in gcf file are not ordered in time (they usually are), cps
 //    2017-05-23   Removed BUG in new merging algorithm in add_GcfSeg that triggered when
-//                  data vectors had not been decoded --- Note, this diables new scheme
+//                  data vectors had not been decoded --- Note, this disables new scheme
```
this is a vendored file not written by us and thus should not be changed for typos IMHO

megies commented on obspy/io/gse2/src/GSE_UTI/gse_functions.c:
```diff
@@ -231,7 +231,7 @@ int decomp_6b_buffer (int n_of_samples, int32_t *dta, char * (* reader)(char *,
   		{fprintf (stderr, "decomp_6b: missing input line?\n"); return -1; }
-      /* We need a space to be sure that CHK2 is not occuring in the middle
+      /* We need a space to be sure that CHK2 is not occurring in the middle
        * of the encoded string/buffer */
```
This is mostly vendored but we made some changes, and I guess the changes can stay in

megies commented on obspy/io/kinemetrics/docs/KW2ASC.SRC:
```diff
@@ -32,7 +32,7 @@
    util.c            Other sub-functions for time decode and Motorola
-                     to Intel byte swaping.
+                     to Intel byte swapping.
 */
```
I believe this is an external resource and I think we should keep it as is

megies commented on obspy/io/seiscomp/data/quakeml_1.2__sc3ml_0.10.xsl:
```diff
@@ -103,7 +103,7 @@ Parent              Element lost
 originUncertainty   confidenceLevel
-arrival             commment
+arrival             comment
 arrival             horizontalSlownessWeight
```
not sure about these stylesheets, I think some are external documents, some probably done by us, maybe keep them unchanged

nathanjmcdougall commented on obspy/io/rg16/core.py:
```diff
@@ -339,7 +339,7 @@ def _read_trace_header_5(fi, trace_block_start):
         '8': 'post processed (HiPAP and INS)',
         '9': 'post processed (HiPAP and DVL)',
-        '10': 'post processed (HiPAP, DVL ans INS)',
+        '10': 'post processed (HiPAP, DVL and INS)',
```
FYI this sort of thing is why I had `ans` -> `prop`. But you're right that `ans` is not strictly a typo.

nathanjmcdougall commented:
Alright, I've tried to do the following:

- Reverted all code changes with the exception of two places (see comments above).
- Documented in `CONTRIBUTING.md` the places where I understand there to be vendored code
- Reverted all changes in those places.

ThomasLecocq commented:
@nathanjmcdougall I propose to close this here for now, and rediscuss this in the future when 1.5 is out, and 👍 
- apply to master
- only to .py files
- by module ?
- and compare e.g. with astral's https://docs.astral.sh/ty/

nathanjmcdougall commented:
Yep, that's fine by me. Not sure about the connection with ty - that's a type checker (like mypy, pyright etc.) for type annotations not for typos/spelling.
